### PR TITLE
Adding HappyBase table helpers.

### PR DESCRIPTION
**Pull request by dhermes**

### CODE REVIEW

File: gcloud/bigtable/happybase/table.py [hunk 1]
```diff
@@ -561,3 +565,135 @@ def _gc_rule_to_dict(gc_rule):
...
+    :param str_val: A string containing the character to update.
+
+    :type index: int
+    :param index: An integer index in ``str_val``.
+
+    :rtype: str
+    :returns: The next character after the character at ``index``
+              in ``str_val``.
+    """
+    if six.PY3:  # pragma: NO COVER
```
Comment (jonparrott) [inline gcloud/bigtable/happybase/table.py]: Why no cover on this little one?

File: gcloud/bigtable/happybase/table.py [hunk 2]
```diff
@@ -561,3 +565,135 @@ def _gc_rule_to_dict(gc_rule):
...
+    :rtype: str
+    :returns: The next string in lexical order after ``str_val``.
+    """
+    str_val = _to_bytes(str_val, encoding='latin-1')
+    if str_val == b'':
+        return str_val
+
+    index = len(str_val) - 1
+    while index >= 0:
+        if six.PY3:  # pragma: NO COVER
```
Comment (jonparrott) [inline gcloud/bigtable/happybase/table.py]: Same here. Are we not running over these on py3?

File: gcloud/bigtable/happybase/table.py [hunk 3]
```diff
@@ -561,3 +565,135 @@ def _gc_rule_to_dict(gc_rule):
...
+    drops everything after it. If the string only contains ``0xFF`` bytes,
+    ``''`` is returned.
+
+    :type str_val: str
+    :param str_val: String to increment.
+
+    :rtype: str
+    :returns: The next string in lexical order after ``str_val``.
+    """
+    str_val = _to_bytes(str_val, encoding='latin-1')
```
Comment (jonparrott) [inline gcloud/bigtable/happybase/table.py]: I'm sure there's a good reason, but I have to ask why `latin-1`?

File: gcloud/bigtable/happybase/table.py (hunk 1 above)
Comment (dhermes) [inline gcloud/bigtable/happybase/table.py]: `tox -e cover` only runs the tests with Python 2.7 (I don't even know how I'd run a single `tox` env. with multiple Python binaries)

File: gcloud/bigtable/happybase/table.py [hunk 4]
```diff
@@ -561,3 +565,135 @@ def _gc_rule_to_dict(gc_rule):
...
+
+    :type cells: list
+    :param cells: List of :class:`.Cell` returned from a read request.
+
+    :type include_timestamp: bool
+    :param include_timestamp: Flag to indicate if cell timestamps should be
+                              included with the output.
+
+    :rtype: list
+    :returns: List of values in the cell. If ``include_timestamp=True``, each
```
Comment (jonparrott) [inline gcloud/bigtable/happybase/table.py]: I would love to see a few examples of return values here.

File: gcloud/bigtable/happybase/table.py (hunk 3 above)
Comment (dhermes) [inline gcloud/bigtable/happybase/table.py]: `latin-1` is for all 8 bits to map onto a byte, `ascii` drops 128 through 255.

File: gcloud/bigtable/happybase/table.py (hunk 2 above)
Comment (dhermes) [inline gcloud/bigtable/happybase/table.py]: Ditto

File: gcloud/bigtable/happybase/table.py (hunk 4 above)
Comment (dhermes) [inline gcloud/bigtable/happybase/table.py]: :+1: Good call

File: gcloud/bigtable/happybase/table.py (hunk 3 above)
Comment (jonparrott) [inline gcloud/bigtable/happybase/table.py]: Gotcha.

File: gcloud/bigtable/happybase/table.py (hunk 1 above)
Comment (jonparrott) [inline gcloud/bigtable/happybase/table.py]: Yay tox. Hmm. I feel like if this is a common theme it should be in `.coveragerc`, not in the code.

Comment (dhermes) [inline gcloud/bigtable/happybase/table.py]: Thanks for persisting on this. I should've have read the `six` docs closer. The fact that

```python
>>> # PY2
>>> (b'ABC')[0]
'a'
>>> # PY3
>>> (b'ABC')[0]
65
```

is what makes this difficult but `six.indexbytes(b'ABC', 0)` produces 65 every time.

**Comment by dhermes** [comment]
@jonparrott PTAL

**Comment by jonparrott** [comment]
LGTM

File: gcloud/bigtable/happybase/table.py (hunk 1 above)
Comment (jonparrott) [inline gcloud/bigtable/happybase/table.py]: Neat, TIL.

**Comment by dhermes** [comment]
Blerg! I merged another fake commit. I need to just go back to writing legitimate commit messages just in case I forget.

**Pull request merged**